src/core/src/factory.rs

```diff
@@ -462,7 +466,7 @@ pub trait Factory<R: Resources> {
         for (rd, d) in raw_data.iter_mut().zip(data.iter()) {
             *rd = cast_slice(*d);
         }
-        self.create_texture_immutable_u8::<T>(kind, &raw_data[.. data.len()])
+        self.create_texture_immutable_u8::<T>(kind, texture::Mipmap::Provided, &raw_data[.. data.len()])
```

Comment:
the API of `create_texture_immutable` should be the same yes `create_texture_immutable_u8` except for the data type

Reply:
What did you mean here?

Reply:
the parameters of `create_texture_immutable` should now be:
```rust
         &mut self,
         kind: texture::Kind,
         mipmap: texture::Mipmap,
         data: &[&[<T::Surface as format::SurfaceTyped>::DataType]])
```